metadata-validator.py

```diff
@@ -0,0 +1,189 @@
+                    logError(element[METADATA_PATH_KEY], "Could not find ASN.1 file", element_filename)
+
+                    error_flag = True
+   
+    errorCheck(error_flag)
+
+    
+    print("\nVeryfing external metadata dependencies")
+    for element in element_lib:
+        if IMPORTS_KEY in element: 
+            for imported in element[IMPORTS_KEY]:
+                if imported[FROM_KEY] not in [os.path.splitext(asn_file)[0] for item in element_lib if ASN1FILES_KEY in item for asn_file in item[ASN1FILES_KEY]]:
+                    logError(element[METADATA_PATH_KEY], "Could not find module for import", imported[FROM_KEY])
+                    error_flag = True
+                    
+    errorCheck(error_flag)
+
+
+    print("\nGenerating and building C files")
+    for element in element_lib:
+        try:
+            [os.remove(x) for x in glob.glob(ASN1SCC_OUT_PATH + "*.c")]
+            [os.remove(x) for x in glob.glob(ASN1SCC_OUT_PATH + "*.h")]
+            [os.remove(x) for x in glob.glob(CC_OUT_PATH + "*.o")]
+            [os.remove(x) for x in glob.glob(AR_OUT_PATH + "*.a")]
+
+        except OSError:
+            print("Error when removing temporary files: %s" (BUILD_PATH))
+            sys.exit(1)
+        
+        dependencies = set(getRecursiveDependency(element_lib, element))
+
+        if len(dependencies):
+            asn1scc_command = ASN1SCC + " " +  ASN1SCC_ARGS + " " + (" ".join(dependencies)) + " -o " + ASN1SCC_OUT_PATH
+
+            print(asn1scc_command)
+            return_code =  os.system(ASN1SCC + " " +  ASN1SCC_ARGS + " " + (" ".join(dependencies)) + " -o " + ASN1SCC_OUT_PATH)
+            if return_code:
+                logError(element[METADATA_PATH_KEY], "Element imported in ASN.1 but not declared in metadata", element[NAME_KEY])
+                error_flag = True
+            
+            else:
+                cfiles = glob.glob(ASN1SCC_OUT_PATH + "*.c")
+                for src in cfiles:
+                    compile_command = CC + " " +  CC_ARGS + " -I " + CC_SRC_PATH + " " + src + " -o " + CC_OUT_PATH + os.path.splitext(os.path.basename(src))[0] + ".o"
+
+                    print(compile_command)
+                    return_code = os.system(compile_command)
+                    if return_code:
+                        logError(element[METADATA_PATH_KEY], "Compilation failed", element[NAME_KEY])
+                        error_flag = True
+                    
+                ar_command = AR + " " + AR_ARGS + " " + AR_OUT_PATH + element[NAME_KEY].replace(" ", "_") + ".a " +  " ".join(glob.glob(CC_OUT_PATH + "*.o"))
+
+                print(ar_command)
+                return_code = os.system(ar_command)
+
+                if return_code:
+                    logError(element[METADATA_PATH_KEY], "Library creation failed", element[NAME_KEY])
+                    error_flag = True
+
+            print()
+    
+    errorCheck(error_flag)
+    sys.exit(0)
+
+main()
```

Comment:
this will execute `main()` even when module will be imported by other module. Standard way for executing main only when script was called directly is such construct:
```
if __name__ == "__main__":
  main()
```

also, consider moving sys.exit here

```
if __name__ == "__name__":
  sys.exit(0 if check() else 1)
```